File: src/create_embedded.py
import os
import json

files = [
    "index.html",
    "style.css",
    "app.js",
    "joystick.js",
    "websocket.js",
    "ui.js"
]

out = ["#pragma once", "", "#include <Arduino.h>", ""]

for f in files:
    path = os.path.join(os.path.dirname(__file__), "..", "data", f)
    with open(path, "r", encoding="utf-8") as file:
        content = file.read()
        
    var_name = f.replace(".", "_")
    
    out.append(f"const char embedded_{var_name}[] PROGMEM = {json.dumps(content)};")
    out.append(f"const size_t embedded_{var_name}_len = sizeof(embedded_{var_name}) - 1;")
    out.append("")

with open(os.path.join(os.path.dirname(__file__), "embedded_files.h"), "w") as file:
    file.write("\n".join(out))

File: src/create_embedded.js
const fs = require('fs');
const path = require('path');

const files = [
    "index.html",
    "style.css",
    "app.js",
    "joystick.js",
    "websocket.js",
    "ui.js",
    "telemetry.js",
    "camera.js"
];

const out = ["#pragma once", "", "#include <Arduino.h>", ""];

for (const f of files) {
    const filePath = path.join(__dirname, "..", "data", f);
    const content = fs.readFileSync(filePath, "utf-8");
    
    const varName = f.replace(/\./g, "_");
    
    out.push(`const char embedded_${varName}[] PROGMEM = ${JSON.stringify(content)};`);
    out.push(`const size_t embedded_${varName}_len = sizeof(embedded_${varName}) - 1;`);
    out.push("");
}

fs.writeFileSync(path.join(__dirname, "embedded_files.h"), out.join("\n"));
console.log("embedded_files.h generated successfully.");
